### PR TITLE
Train reward-shaping model with large data

### PR DESCRIPTION
**Pull request by nissymori**
#1127

### CODE REVIEW

**Comment by nissymori** [comment]
データ数: 17369
learning rate: 0.001
epoch数: 50
batch size 16
で学習させて, 局数と自分の点数を変化させてプロットしてみました. 局ごとの変化がないのが気になるので, 調査しようと思います.

スクリーンショット 2022-09-04 19 46 10

**Comment by nissymori** [comment]
データ数, learning rateは同じで, epoch数を200, batch sizeを32にすると割と局ごとの差も出て良い感じな気がします.

スクリーンショット 2022-09-04 20 04 03

**Comment by sotetsuk** [comment]
局もonehotを加えるか置き換えるかしてみてもいいかも？

**Comment by sotetsuk** [comment]
あとは学習曲線がどんな感じかにもよる気がします

**Comment by nissymori** [comment]
了解です！

**Comment by sotetsuk** [comment]
あとたぶんデータ全然足りない気がしますね

**Comment by sotetsuk** [comment]
Epoch数50はもう結果を覚えてるでしょ。バリデーションセット用意しました？

**Comment by nissymori** [comment]
> Epoch数50はもう結果を覚えてるのでは？バリデーションセット用意しました？

とりあえずlossが下がるかを確認するためだったので, 用意していません.
データ増やしてvalidation setも用意してやってみようと思います！

**Comment by nissymori** [comment]
ToDo

- [ ] plot learning curve
- [ ] increase data
- [ ] validation

**Comment by nissymori** [comment]
validation dataを追加して訓練してみました.
learning rate 0.002, epoch 30, バッチサイズ 64です.

データ数はまだ32000ほどです.

スクリーンショット 2022-09-08 15 11 01
スクリーンショット 2022-09-08 15 13 30

**Comment by nissymori** [comment]
- [  ] Readme書く.
- [  ] データ数を増やす.

**Comment by nissymori** [comment]
南四局と途中で終わっている場合は最終局のみのデータを使って学習させた結果と学習曲線です.
epoch数は10です.

局をランダムに選んだ場合とあまり違いがない様に思います. 
一旦10万くらいまでデータを増やしてみます.

スクリーンショット 2022-09-09 11 45 48
スクリーンショット 2022-09-09 11 45 33

**Comment by nissymori** [comment]
試しに1000epochくらい学習させてみました.
直感に近づいている気がします. val lossは悪化していますが.

とりあえずデータ増やしてみます.



スクリーンショット 2022-09-09 12 01 34


スクリーンショット 2022-09-09 12 01 24

**Comment by nissymori** [comment]
変更点
- 特徴量として残りの親の数を追加する.
- 出力は4次元[起家, ..ラス親]
- NNは8個用意する[model_0, ...model_7](東1から南4)

procedure
-  南4のデータから順位点をmodel_7に予測させる
- 南３のデータのmodel_7の予測結果をmodel_6に予測させる.
- 以下繰り返す.

**Comment by nissymori** [comment]
## 特徴量再考
出力を四次元にするのでそれに伴って特徴量設計を変えた方が良い.

- 4人分の点 [起家, ..ラス親]の順で与える.(今まではtargetを起点にしていた)
- 本場: 変更なし
- 詰み棒: 変更なし
- 局: 変更なし
- 風: one-hotで四人分? 起家の風だけone-hotでもいい気がする.
- 親: one-hot
- 残りの親の数: 起家から順番に

**Comment by nissymori** [comment]
長くなってきたので, 一旦このPR閉じてしまってもよろしいでしょうか.

**Comment by sotetsuk** [comment]
Typo suphx

マージしちゃって下さい！@nissymori

**Comment by nissymori** [comment]
> Typo suphx（suphnxがある）
> 
> マージしちゃって下さい！@nissymori

ありがとうございます！

typo直してmergeします！
